src/traces/violin/calc.js

```diff
@@ -42,13 +42,12 @@ module.exports = function calc(gd, trace) {
 
         // sample standard deviation
         var ssd = Lib.stdev(vals, len - 1, cdi.mean);
-        var bandwidthDflt = ruleOfThumbBandwidth(vals, ssd, cdi.q3 - cdi.q1);
-        var bandwidth = cdi.bandwidth = trace.bandwidth || bandwidthDflt;
+        var bandwidth = cdi.bandwidth = trace.bandwidth || ruleOfThumbBandwidth(vals, ssd, cdi.q3 - cdi.q1);
         var span = cdi.span = calcSpan(trace, cdi, valAxis, bandwidth);
 
         // step that well covers the bandwidth and is multiple of span distance
         var dist = span[1] - span[0];
-        var n = Math.ceil(dist / (Math.min(bandwidthDflt, bandwidth) / 3));
+        var n = Math.ceil(dist / (bandwidth / 3));
```

Comment:
Any reason for choosing just `bandwidth` here instead of `bandwidthDflt ? Math.min(bandwidthDflt, bandwidth) : bandwidth`?

Reply:
Ah ok. In this case, I'm thinking capping `n` would be best. Probably `n = Math.max(n, 1e5)` should suffice.

Reply:
> Probably `n = Math.max(n, 1e5)` should suffice

You mean `Math.min`, but agreed. As is (with or without this PR) you could crash the browser just by setting a small bandwidth. @etpinard was there a reason to use `Math.min(bandwidthDflt, bandwidth)` before? Seems like 3 points per bandwidth will get the curve shape correct even if you set a large explicit bandwidth, won't it?

Reply:
> was there a reason to use Math.min(bandwidthDflt, bandwidth) before?

Good question. It comes from the very first violin [commit](https://github.com/plotly/plotly.js/commit/3438eae9199ef5df36a907ccb3b8b54ad16fcd6e#diff-566957ce4e618a69e42f7faaa18dec12R37). I must have taken that (possibly erroneously) from the seaborn implementation.